watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack.py

```diff
@@ -0,0 +1,326 @@
+    m.fs.EDstack.cell_width.fix(0.1)
+    m.fs.EDstack.cell_length.fix(0.79)
+    m.fs.EDstack.membrane_thickness["aem"].fix(1.3e-4)
+    m.fs.EDstack.membrane_thickness["cem"].fix(1.3e-4)
+    m.fs.EDstack.solute_diffusivity_membrane["cem", "Na_+"].fix(1.8e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["aem", "Na_+"].fix(1.25e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["cem", "Cl_-"].fix(1.8e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["aem", "Cl_-"].fix(1.25e-10)
+    m.fs.EDstack.ion_trans_number_membrane["cem", "Na_+"].fix(1)
+    m.fs.EDstack.ion_trans_number_membrane["aem", "Na_+"].fix(0)
+    m.fs.EDstack.ion_trans_number_membrane["cem", "Cl_-"].fix(0)
+    m.fs.EDstack.ion_trans_number_membrane["aem", "Cl_-"].fix(1)
+
+    # check degrees of freedom
+    if degrees_of_freedom(m) != 0:
+        raise RuntimeError(
+            "The set_operating_conditions function resulted in {} "
+            "degrees of freedom rather than 0. This error suggests "
+            "that too many or not enough variables are fixed for a "
+            "simulation.".format(degrees_of_freedom(m))
+        )
+
+
+def solve(blk, solver=None, tee=False, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(blk, tee=True)
+    if check_termination:
+        assert_optimal_termination(results)
+    return results
+
+
+def initialize_system(m, solver=None):
+
+    # set up solver
+    if solver is None:
+        solver = get_solver()
+    optarg = solver.options
+
+    # populate intitial properties throughout the system
+    m.fs.feed.initialize(optarg=optarg)
+    propagate_state(m.fs.s01)
+    m.fs.separator.initialize(optarg=optarg)
+    propagate_state(m.fs.s02)
+    propagate_state(m.fs.s03)
+    m.fs.EDstack.initialize(optarg=optarg)
+    propagate_state(m.fs.s04)
+    m.fs.product.initialize()
+    propagate_state(m.fs.s05)
+    m.fs.disposal.initialize()
+    m.fs.costing.initialize()
+
+
+def optimize_system(m, solver=None):
+
+    # Below is an example of optimizing the operational voltage and cell pair number (which translates to membrane use)
+    # Define a system with zero dof
+    set_operating_conditions(m)
+
+    # Set an objective function
+    m.fs.objective = Objective(expr=m.fs.costing.LCOW)
+
+    # Choose and unfix variables to be optimized
+    m.fs.EDstack.voltage_applied[0].unfix()
+    m.fs.EDstack.cell_pair_num.unfix()
+    # Give narrower bounds to optimizing variables if available
+    m.fs.EDstack.voltage_applied[0].setlb(0.01)
+    m.fs.EDstack.voltage_applied[0].setub(60)
+    m.fs.EDstack.cell_pair_num.setlb(1)
+    m.fs.EDstack.cell_pair_num.setub(1000)
+
+    # Set a treatment goal
+    # Example here is to reach a final product water containing NaCl < 1 g/L (from a 10 g/L feed)
+    m.fs.eq_product_quality = Constraint(
+        expr=m.fs.product.properties[0].conc_mass_phase_comp["Liq", "Na_+"] <= 0.393
+    )
```

Comment:
This comment says less than 1 g/L, but the constraint is less than 0.393 g/L right? Ohhh its Na vs NaCl. You could just add the concentration of Na to the concentration of Cl and set it.

Reply:
Did this.